### PR TITLE
Zenith: Adds descriptions & comments for Python SDK init

### PR DESCRIPTION
**Pull request by jpadams**
To match https://github.com/dagger/dagger/pull/6720 and #6742 

Builds on  https://github.com/dagger/dagger/pull/6729

`ruff format` seems happy.

Similar to Go and TS, I'm not putting a comment on the Class.
With `pylint` that throws

```
src/main.py:19:0: C0115: Missing class docstring (missing-class-docstring)
```

which someone could quiet down with something like this in their `.pylintrc`:
```
[MASTER]
disable=
    C0115, #missing-class-docstring
```

### CODE REVIEW

**Comment by sipsma** [comment]
@jpadams Python SDK lint error seems legit: https://github.com/dagger/dagger/actions/runs/8055979358/job/22004049415?pr=6741#step:6:596 (just need an extra newline before the closing `"""` I think)

The other errors are unrelated, will be fixed by https://github.com/dagger/dagger/pull/6745

**Comment by jpadams** [comment]
> @jpadams Python SDK lint error seems legit: https://github.com/dagger/dagger/actions/runs/8055979358/job/22004049415?pr=6741#step:6:596 (just need an extra newline before the closing `"""` I think)
> 
> The other errors are unrelated, will be fixed by #6745

Got it. Also fixed the error orign on line 600 where `ruff` and others want an extra line.

**Review by sipsma — APPROVED** [review]
LGTM, but better if @helderco signs off too of course before merge

**Comment by sipsma** [comment]
I just merged the other PR to fix docs lints, will rebase this one and let CI run to be extra careful and merge after if all green

**Pull request merged**
